src/veh_interact.h

```diff
@@ -172,6 +172,9 @@ class veh_interact
         void allocate_windows();
         void do_main_loop();
         void deallocate_windows();
+
+        /*Filter stuff*/
+        std::string filter;
```

Comment:
Well you only need that state whilst your filter runs. Try and keep as much as possible out of the public header